### ISSUE TITLE
Unable to update flag between lanes

### ISSUE THREAD

**Issue by matibzurovski** [fastlane/fastlane#11912]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

I am trying to execute multiple lanes from my CI. There is a particular command that I need to execute only the first time, and if I add it inside the `before_all do` statement then it is executed before each lane is ran.

Therefore, I have tried setting up a flag that would be updated after its first usage, but it isn't working. This is how it looks like

```Ruby
customFlag = nil

platform :ios do
    before_all do
      if !customFlag 
        customFlag = "true"
        one_time_lane
      end
  
    end

    lane :one_time_lane do
      put "You should see this message only once"
    end

    lane :lane_a do
      put "lane A"
    end

    lane :lane_b do
      put "lane B"
    end

end
```

Then, if I run the following commands from my CI configuration:
```
bundle exec fastlane lane_a
bundle exec fastlane lane_b
```
I get the following output:
```
You should see this message only once
lane A
You should see this message only once // wrong!
lane B
```

However, the message is printed twice. I see two paths that could fix my issue:
- Have the `customFlag` be properly updated
- Replace `before_all` for some similar statement that executes the block once only, and not one time per lane (couldn't find the solution in https://github.com/fastlane/fastlane/issues/116)

**Comment by taquitos** [comment]
remove the `var` in front of `var customFlag = nil`,  that should work 🚀

**Comment by matibzurovski** [comment]
Sorry @taquitos, that was my swifty fingers tipping for me 😅 

I didn't actually include the `var` keyword on the `Fastfile`, updated the original comment.

**Comment by taquitos** [comment]
Ok, so what's happening here is that you're calling:
 `bundle exec fastlane lane_a` and then
`bundle exec fastlane lane_b`

These are 2 different executions, so you're basically spinning up fastlane 2 times.

You can call multiple lanes from a single lanes, and you won't need to use `customFlag` at all (because `before_all` will only run once)
```ruby
lane :lane_a do
  put "lane A"
end

lane :lane_b do
  put "lane B"
end

lane :both_lanes do
  lane_a
  lane_b
end
```
`bundle exec fastlane both_lanes`

If that isn't what you want, you can use an environment variable which you set after your first fastlane call, and then in the `before_all` block you can check for it in ENV["MyEnvVar"]

**Comment by matibzurovski** [comment]
I see, so variables aren't persisted between different lanes.

Grouping the two calls on one execution won't work, as I run some other steps between them. I have also tried the `ENV["customFlag"] = nil` approach, but the result is the same. You say this should work even if doing one single execution?

**Comment by taquitos** [comment]
So, variables are persisted between lanes as long as it's in the parent scope (like you have it), it's just that in your case, you're actually spinning up 2 processes, so the state is getting reset. That fastfile is getting parsed and executed 2 times, each in a different process spaces.
